pkg/api/types.go

```diff
@@ -76,6 +76,14 @@ type CertificateProfile struct {
 	KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
 	// caPrivateKey is an internal field only set if generation required
 	caPrivateKey string
+	// SwarmTlsServerCertificate is the Docker server daemon certificate and signed by the CA
```

Comment:
CertificateProfile also contains the following which is K8s specific:

```
// KubeConfigCertificate is the client certificate used for kubectl cli and signed by the CA
KubeConfigCertificate string `json:"kubeConfigCertificate,omitempty"`
// KubeConfigPrivateKey is the client private key used for kubectl cli and signed by the CA
KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
```
SwarmMode cert and Kube config certs live side by side in one struct? Its confusing about what is the right combination. Should the CertificateProfile be renamed to KubernetesCertificateProfile and another struct created called "SwarmModeCertificateProfile". That way there can be validation check that one only type is set in properties based on orchestrator profile.